### PR TITLE
Add pre-commit config

### PR DESCRIPTION
**Pull request by shenef**
requires `pip install pre-commit` and `pre-commit install` to install

### CODE REVIEW

**Comment by MrTyton** [comment]
Maybe also use https://github.com/marketplace/actions/autoyapf ?

**Comment by shenef** [comment]
> Maybe also use https://github.com/marketplace/actions/autoyapf ?
That very much looks like it would conflict with black since they are doing the same thing.

**Comment by shenef** [comment]
Code has finished a full run before merging the latest code from main.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+repos:
+-   repo: https://github.com/psf/black
+    rev: 22.10.0
+    hooks:
+    -   id:   black
+        exclude: migrations
```
Comment (jlward) [inline .pre-commit-config.yaml:6]: This isn't needed. This is likely legacy from you copying this from my Django app.

File: save_sphere.py [hunk 2]
```diff
@@ -177,32 +202,31 @@ def record_save_sphere(x_val:int, y_val:int, diag_prog:int, actor:int):
...
-                    records[map_num][diag_num][actor_num]["x"] = newVal[map_num][diag_num][actor_num]["x"]
-                    records[map_num][diag_num][actor_num]["y"] = newVal[map_num][diag_num][actor_num]["y"]
-                    records[map_num][diag_num][actor_num]["diag"] = newVal[map_num][diag_num][actor_num]["diag"]
+                if (
+                    records[map_num][diag_num][actor_num]["diag"]
+                    != newVal[map_num][diag_num][actor_num]["diag"]
+                ):
+
+                    records[map_num][diag_num][actor_num]["x"] = newVal[map_num][
+                        diag_num
```
Comment (jlward) [inline save_sphere.py:216]: I hate how `black` formats so of this stuff, but it's the price we pay for using `black`. It's better to have it than to not have it.

**Review by jlward — APPROVED** [review]

**Comment by jlward** [comment]
We either need to add `pre-commit` to `requirements.txt` or `dev-requirements.txt` and likely also add a note in the README about how to hook it in (running `pre-commit install`)

**Pull request merged**